process_linux.go

```diff
@@ -236,5 +236,14 @@ func (p *initProcess) createNetworkInterfaces() error {
 }
 
 func (p *initProcess) signal(s os.Signal) error {
-	return p.cmd.Process.Signal(s)
+	return maybeSyscall(p.cmd.Process, s)
+}
+
+func maybeSyscall(process *os.Process, s os.Signal) error {
```

Comment:
Not sure if I follow.  I'd prefer we just always use syscall.Kill but because of the type assertion I fell back to process.Signal.  Is it possible to get a os.Signal that is not a syscall.Signal?